credentials/jwt/jwt_file_reader_test.go

```diff
@@ -0,0 +1,172 @@
+		wantErr  error
+	}{
+		{
+			name:     "nonexistent_file",
+			create:   false,
+			contents: "",
+			wantErr:  errTokenFileAccess,
+		},
+		{
+			name:     "empty_file",
+			create:   true,
+			contents: "",
+			wantErr:  errJWTValidation,
+		},
+		{
+			name:     "file_with_whitespace_only",
+			create:   true,
+			contents: "   \n\t  ",
+			wantErr:  errJWTValidation,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var tokenFile string
+			if !tt.create {
+				tokenFile = "/does-not-exist"
+			} else {
+				tokenFile = writeTempFile(t, "token", tt.contents)
+			}
+
+			reader := jwtFileReader{tokenFilePath: tokenFile}
+			if _, _, err := reader.readToken(); err == nil {
+				t.Fatal("ReadToken() expected error, got nil")
+			} else if !errors.Is(err, tt.wantErr) {
+				t.Fatalf("ReadToken() error = %v, want error %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func (s) TestJWTFileReader_ReadToken_InvalidJWT(t *testing.T) {
+	now := time.Now().Truncate(time.Second)
+	tests := []struct {
+		name         string
+		tokenContent string
+		wantErr      error
+	}{
+		{
+			name:         "valid_token_without_expiration",
+			tokenContent: createTestJWT(t, time.Time{}),
+			wantErr:      errJWTValidation,
+		},
+		{
+			name:         "expired_token",
+			tokenContent: createTestJWT(t, now.Add(-time.Hour)),
+			wantErr:      errJWTValidation,
+		},
+		{
+			name:         "malformed_JWT_not_enough_parts",
+			tokenContent: "invalid.jwt",
+			wantErr:      errJWTValidation,
+		},
+		{
+			name:         "malformed_JWT_invalid_base64",
+			tokenContent: "header.invalid_base64!@#.signature",
+			wantErr:      errJWTValidation,
+		},
+		{
+			name:         "malformed_JWT_invalid_JSON",
+			tokenContent: createInvalidJSONJWT(t),
+			wantErr:      errJWTValidation,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tokenFile := writeTempFile(t, "token", tt.tokenContent)
+
+			reader := jwtFileReader{tokenFilePath: tokenFile}
+			if _, _, err := reader.readToken(); err == nil {
+				t.Fatal("ReadToken() expected error, got nil")
+			} else if !errors.Is(err, tt.wantErr) {
+				t.Fatalf("ReadToken() error = %v, want error %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func (s) TestJWTFileReader_ReadToken_ValidToken(t *testing.T) {
+	now := time.Now().Truncate(time.Second)
+	tokenExp := now.Add(time.Hour)
+	token := createTestJWT(t, tokenExp)
+	tokenFile := writeTempFile(t, "token", token)
+
+	reader := jwtFileReader{tokenFilePath: tokenFile}
+	readToken, expiry, err := reader.readToken()
+	if err != nil {
+		t.Fatalf("ReadToken() unexpected error: %v", err)
+	}
+
+	if readToken != token {
+		t.Errorf("ReadToken() token = %q, want %q", readToken, token)
+	}
+
+	if !expiry.Equal(tokenExp) {
+		t.Errorf("ReadToken() expiry = %v, want %v", expiry, tokenExp)
+	}
+}
+
+// createInvalidJSONJWT creates a JWT with invalid JSON in the payload.
+func createInvalidJSONJWT(t *testing.T) string {
```

Comment:
nit: We can omit JSON from the function name since the J in JWT stands for JSON.